### ISSUE TITLE
[Tracking][Image][Polish] Image-related feature request and improvements

### ISSUE THREAD

**Issue by roryabraham** [Expensify/App#10894]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

### Problem
We have seen a number of issues come up related to images. Uploading them, editing them, making them available offline, copying them to clipboard, etc...

Onyx _does_ have support for images, but AFAIK nobody has utilized that functionality to build any app features that follow our offline-first paradigm for images.

Overall, I think there's a lot of confusion around this topic (speaking for myself here, so feel free to comment if you disagree), and that leads to potentially disparate approaches or complex code surrounding images.

### Solution
As these come up, we see the need for a common, unified approach to handling images, presumably using Onyx. This issue is meant to serve as a tracking issue for all image-related feature requests and bug fixes. We should look for commonality between these features, try to come up with a generalized, future-proof solution that's easy to work with, and implement it via the WN/Design Doc process.

#### Setting up Caching Issue(s)
- [x] https://github.com/Expensify/App/issues/10792
- [x] https://github.com/Expensify/App/issues/12603
- [x] https://github.com/Expensify/App/issues/9703
    - [x] https://github.com/Expensify/App/issues/12259
- [x] https://github.com/Expensify/App/issues/6527

#### Bugs based off caching [Holding on ^]
- [x] https://github.com/Expensify/App/issues/12185
- [x] https://github.com/Expensify/App/issues/11425
- [x] https://github.com/Expensify/App/issues/12707

#### "Flashing" / Image rerendering [Holding on ^]
- [ ] https://github.com/Expensify/App/issues/9402
- [x] https://github.com/Expensify/App/issues/10704
- [x] https://github.com/Expensify/App/issues/11393

#### New Features
- [ ] https://github.com/Expensify/App/issues/4629
- [x] https://github.com/Expensify/App/issues/7862
- [x] https://github.com/Expensify/App/issues/11085
- [x] https://github.com/Expensify/App/issues/11197
- [ ] Use Blurhash as image placeholders while they're loading ([slack convo](https://expensify.slack.com/archives/C035J5C9FAP/p1675427324162769))
- [x] Offline Image Support - https://github.com/Expensify/App/issues/17069

#### Bugs (in progress - not on hold)
- [x] Default avatars: Use SVG asset that we can change the color
    - `FallbackAvatar` & `FallbackWorkspaceAvatar`
    - Note: Being handled in this issue: https://github.com/Expensify/App/issues/12259
- [x] https://github.com/Expensify/App/issues/15288
- [x] https://github.com/Expensify/App/issues/11063
    - [x] https://github.com/Expensify/App/issues/13038
- [x] https://github.com/Expensify/App/issues/13006
- [x] https://github.com/Expensify/App/issues/13555
- [x] https://github.com/Expensify/App/issues/13585
- [x] https://github.com/Expensify/App/issues/13962
- [x] https://github.com/Expensify/App/issues/14027

### Resolved issues
- [x] https://github.com/Expensify/App/issues/10029
- [x] https://github.com/Expensify/App/issues/10849
- [x] https://github.com/Expensify/App/issues/9418
- [x] https://github.com/Expensify/App/issues/9239
- [x] https://github.com/Expensify/App/issues/10258
- [x] https://github.com/Expensify/App/issues/8846
- [x] https://github.com/Expensify/App/issues/11639
- [x] https://github.com/Expensify/App/issues/10950
- [x] https://github.com/Expensify/App/issues/8235
- [x] https://github.com/Expensify/App/issues/10599
- [x] https://github.com/Expensify/App/issues/11854
- [x] https://github.com/Expensify/App/issues/13300
- [x] https://github.com/Expensify/App/issues/11511

Follow-up discussions:
- [Stop storing full image URLs in the database](https://expensify.slack.com/archives/C01GTK53T8Q/p1665588902025629)
    - Decided this wasn't needed for "WhatsApp Quality" but still useful to do eventually

**Comment by roryabraham** [comment]
A project would be fine – historically we've usually just used issues but a project board could be useful. More importantly, we need someone to take this on as a project manager.

One interesting idea could be to engage [@thomas-coldwell](https://github.com/thomas-coldwell) from Margelo in this project, since [@mrousavy](https://github.com/mrousavy) told me that Thomas typically works on Video/Audio/Image/Graphics stuff. Thoughts @mrousavy?

**Comment by mrousavy** [comment]
@roryabraham yes, this would probably pretty interesting! 😄 

I'll talk to Thomas about availability then get back to you here or on Slack :)

**Comment by roryabraham** [comment]
Added https://github.com/Expensify/App/issues/9703 to this list.

**Comment by roryabraham** [comment]
There was some useful discussion in slack regarding this issue [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1664654201417359). Added @kidroca's summary to the OP for context

**Comment by Beamanator** [comment]
Added https://github.com/Expensify/App/issues/11425 to the list

**Comment by Beamanator** [comment]
Finished looking through existing issues & adding them to this tracker, tomorrow I'll comb through them to figure out what common things we need to implement & some ideas how to do that, before proceeding

**Comment by JmillsExpensify** [comment]
Asked a question about next steps [in Slack](https://expensify.slack.com/archives/C01GTK53T8Q/p1665437694463779). Feel free to answer here or there. No need to answer in both places, thanks!

**Comment by mallenexpensify** [comment]
:whoohooo: just scored me 17 open issues to BZ! 🏅 
Need to develop a process for this and add/update SO.  Will do soon

**Comment by Beamanator** [comment]
> :whoohooo: just scored me 17 open issues to BZ! 🏅
Need to develop a process for this and add/update SO. Will do soon

Hahaha glad you're happy about this! Lots of work to do 💪

**Comment by Beamanator** [comment]
Sent a pre-design recap on Monday [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1666627315314999).

Today's goal: Get Web-E to accept encrypted auth token via `header` (instead of via the image URL query param). Once this is possible, we (hopefully Kidroca or a Margelo dev) can whip up some POCs to make sure the latest plan ([here](https://expensify.slack.com/archives/C01GTK53T8Q/p1666716374309799?thread_ts=1666627335.351399&cid=C01GTK53T8Q)) is possible

**Comment by mallenexpensify** [comment]
Chatted this morn!  We might want to update the list in the OP to bunch issues/themes or if some issues are help on others.

**Comment by Beamanator** [comment]
Reorganized the issue list a bit by categorizing open issues - it looks like some of the issues "On hold" (mainly bug fixes) can be taken off hold if they don't have anything to do with image flashing / image caching

**Comment by mallenexpensify** [comment]
> It looks like some of the issues "On hold" (mainly bug fixes) can be taken off hold if they don't have anything to do with image flashing / image caching

So... the bottom three here @Beamanator ? 
<img width="849" alt="image" src="https://user-images.githubusercontent.com/22508304/199124916-2833180f-8ac9-4490-9109-224cf8c341d8.png">

**Comment by Beamanator** [comment]
Ya I think all 4 @mallenexpensify - the top one is just about the zoom feature, so probably doesn't need to wait on anything

**Comment by mallenexpensify** [comment]
Updated all issues in the `Bugs (in progress - not on hold)` section and took off hold which is now reflected in the OP

**Comment by Beamanator** [comment]
Added https://github.com/Expensify/App/issues/11854 to `Bugs (in progress - not on hold)`

**Comment by Beamanator** [comment]
Posted in #whatsnext end of last week ([here](https://expensify.slack.com/archives/CC7NECV4L/p1667578884362449))

Will create implementation issues tomorrow if there's no changes in the solution path!

**Comment by mallenexpensify** [comment]
Co-assigning @trjExpensify as BZ, he'll be taking the BZ lead on this and all related issues til I'm back, hopefully in a month.

**Comment by Beamanator** [comment]
Current top priority issues for this proj:
- https://github.com/Expensify/App/issues/10792
- https://github.com/Expensify/App/issues/12603
- https://github.com/Expensify/App/issues/11063

**Comment by JmillsExpensify** [comment]
Hugely helpful. Keep these kinds of hit lists coming!

**Comment by trjExpensify** [comment]
No change to [the hit list](https://github.com/Expensify/App/issues/10894#issuecomment-1311119746) from EOD Friday, Melv.

**Comment by Beamanator** [comment]
Adding new issue to the list: https://github.com/Expensify/App/issues/12707

I think this one should be placed on hold, it should be solved once caching is implemented

**Comment by trjExpensify** [comment]
Evening, Melvin! 

- https://github.com/Expensify/App/pull/12648#issuecomment-1317505702 is with @mananjadhav for review
- https://github.com/Expensify/App/issues/12603#issuecomment-1314276420 - is with @kidroca to fix `react-native-web` for web/desktop image caching
- https://github.com/Expensify/App/pull/12549 - is on @Beamanator to address @Santhosh-Sellavel's first review comments from earlier today. Alex is out for the rest of the week, so this will slow down now until Monday.

**Comment by Beamanator** [comment]
Good progress on all of the above ^ PRs, hoping to get them all knocked out this week (and I'm back so I'll be helping push things forward 🙏 )

**Comment by trjExpensify** [comment]
- Fast Image for native is merged 🎉 
- RNW Image for web/desktop is [now ready](https://expensify.slack.com/archives/C01GTK53T8Q/p1669275909710189?thread_ts=1669058284.677789&cid=C01GTK53T8Q) to be reviewed? 
- Allow multiple avatar image type has been reviewed by C+, but awaiting comments to be addressed.

**Comment by Beamanator** [comment]
Yep yep thanks for the update 👍  Notes:
- Fast Image PR may have caused a regression 🙃 (being investigated [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1669305509853359))
- RNW caching has been updated and "should work" but will be tested more soon (discussion [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1669308015285649?thread_ts=1669058284.677789&cid=C01GTK53T8Q))
- Allowing more avatar image types - I ran into some potential issues with 3rd party libraries we use, I am thinking about trying to move forward with "what we currently can do" & make follow-up issue for allowing `.bmp` and `.gif` images to work in Android & iOS

**Comment by JmillsExpensify** [comment]
> Allowing more avatar image types - I ran into some potential issues with 3rd party libraries we use, I am thinking about trying to move forward with "what we currently can do" & make follow-up issue for allowing .bmp and .gif images to work in Android & iOS

I like this plan.

**Comment by Beamanator** [comment]
Updates for the day:
- Fast Image PR (the one that was reverted): was reviewed by `@kidroca` with some useful comments, looking forward to the follow-up PR from @thomas-coldwell 
- RNW caching:
    - Looks like `@necolas` (maintainer of `react-native-web`) is responding to the PR - though they might have preferred we talk to them _before_ creating the PR ([here](https://github.com/necolas/react-native-web/pull/2442#issuecomment-1329664670))
    - I'm still not able to see the new header on the backend, meaning I don't see images loading properly on web yet 😭 
- [More image types](https://github.com/Expensify/App/pull/12549): Getting real close, still two weird behaviors (one NAB and I thinkkk one caused by a browser setting)

**Comment by Beamanator** [comment]
Quick updates:
1. Fast Image PR: Looks like we're still waiting on Thomas to get better, I haven't seen a draft PR yet
2. RNW caching: Latest updates [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1669910230049609?thread_ts=1669058284.677789&cid=C01GTK53T8Q)
    - Basically, kidroca & the `react-native-web` maintainer look to be pushing toward getting kidroca's PR (https://github.com/necolas/react-native-web/pull/2442) merged!
    - Not entirely sure how long this will take, but the current plan is hold off on the Fork PR for a bit, hope ^ gets merged, then keep moving
3. [More image types](https://github.com/Expensify/App/pull/12549): Down to the last test to see if iOS mWeb (safari) has bugs or is working well -> Investigating if there's a difference between iOS 15.5 (my sim versions) & iOS 16.1 (Santhosh's device version)

**Comment by Beamanator** [comment]
Updates:
1. Fast Image PR: Looks like Thomas is getting round to reviewing comments on his previous PR!
2. RNW caching: 
3. [More image types](https://github.com/Expensify/App/pull/12549) is merged!
    - Once this makes it to Production (assuming there's no regressions) I'll take https://github.com/Expensify/App/issues/13038 off hold and move forward there!
4. [Default Avatars](https://github.com/Expensify/App/issues/9703): Wanted to start tracking this since it's another main caching issue. 
    - As Cristi mentioned [here](https://github.com/Expensify/App/issues/9703#issuecomment-1334522208), Georgia will be continuing this effort since she has been working on default avatar updates a lot lately - she already has some PRs up (see comment) and will modify those to use our latest solution (around [here](https://github.com/Expensify/App/issues/9703#issuecomment-1333977509))

**Comment by Beamanator** [comment]
Updates:
1. Fast Image PR: Thomas has a draft (https://github.com/Expensify/App/pull/13304), but we're deciding how to proceed since the regression was actually a bug in a 3rd party lib, and we want to fix upstream if possible
2. RNW caching: Waiting on convo in 3rd party lib (waiting for maintainer to review upstream PR)
3. Multiple image types:
    - Made a new follow-up to look into improving the background in the avatar crop modal, as black background doesn't look so great: https://github.com/Expensify/App/issues/13266
    - Main PR is [in staging](https://github.com/Expensify/App/pull/12549#issuecomment-1335743138), will wait another day or 2 before taking https://github.com/Expensify/App/issues/13038 off hold
4. Default avatars: no update

**Comment by trjExpensify** [comment]
>RNW caching: Waiting on convo in 3rd party lib (waiting for maintainer to review upstream PR)

@Beamanator mind linking this? Also broadly, what's the latest here?

**Comment by Beamanator** [comment]
Was OOO, will add an update here tomorrow for sure 🙏

**Comment by Beamanator** [comment]
Updates:
- iOS / Android (`react-native-fast-image`): 
    - Started a convo [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1671112210394559?thread_ts=1670868002.066289&cid=C01GTK53T8Q) to see if we want to implement a small workaround that can unblock his main caching PR, or if we want to wait for an upstream fix ([here](https://github.com/Expensify/App/pull/13304#issuecomment-1352897653) Thomas said he's starting that PR) before moving forward
    - His new caching PR (draft) is here: https://github.com/Expensify/App/pull/13304
- Web / Desktop (`react-native-web`):
    - Kidroca's PR (https://github.com/necolas/react-native-web/pull/2442) is moving pretty slow, the maintainer mentioned [here](https://github.com/Expensify/App/issues/12603#issuecomment-1343269836) that they're busy working on many other things for the next release
    - I may start a convo soon about implementing the update in our fork while waiting for upstream progress
- Multiple avatar image types:
    - Next target issue: https://github.com/Expensify/App/issues/13038
    - A few convos we need to align on (coming soon)
    - My changed caused this issue: https://github.com/Expensify/App/issues/13555
- [Default avatars](https://github.com/Expensify/App/issues/12259)
    - Update from Georgia here: https://github.com/Expensify/App/issues/12259#issuecomment-1346937345

**Comment by trjExpensify** [comment]
>I may start a convo soon about implementing the update in our fork while waiting for upstream progress

In light of [the discussion](https://expensify.slack.com/archives/C01GTK53T8Q/p1671128570739449?thread_ts=1670868002.066289&cid=C01GTK53T8Q) we had for the process with forks, I wouldn't delay this any longer. This is really holding us up on making further progress with the Image improvements project.

**Comment by trjExpensify** [comment]
@Beamanator thoughts on the above?

**Comment by Beamanator** [comment]
Yes good call, I can definitely start the convo tomorrow - sorry for the delay, today I was cut short due to some personal issues but will get that going tomorrow 💪

**Comment by Beamanator** [comment]
Updates:
- iOS  / Android:
    - Upstream PR: https://github.com/DylanVann/react-native-fast-image/pull/953
    - Plan if it doesn't move along is here: https://github.com/Expensify/App/issues/10792#issuecomment-1358020140
- Web / Desktop:
    - Upstream PR: https://github.com/necolas/react-native-web/pull/2442
    - Hasn't had updates in a bit, so we have a [new plan](https://github.com/Expensify/App/issues/12603#issuecomment-1359590914):
        - Get the fix merged into our `react-native-web` fork (PR: https://github.com/Expensify/react-native-web/pull/8)
        - Get it updated in App, so we can unblock multiple caching issues
        - Create new `Monthly` tracking issue to keep up with status of upstream PR
- Multiple avatar image types
    - Posted in expensify-open-source [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1671547890266159) to align on next steps for https://github.com/Expensify/App/issues/13038#issuecomment-1359504509
- Default avatars (https://github.com/Expensify/App/issues/12259):
    - No updates

**Comment by JmillsExpensify** [comment]
Note for next week: Reach out to maintainer for #953 and ask if we have an ETA for potentially merging.

**Comment by JmillsExpensify** [comment]
As a heads up, I've removed this issue from the WAQ project and filed it in N7 instead. If you're curious for more context, it's captured in this [Slack thread](https://expensify.slack.com/archives/C01GTK53T8Q/p1672088203968479). Summarizing that convo though, this mainly comes down to us moving the big architectural changes into whatever respective projects they block. In the case of images, that's minimally when we get users interacting with their company and sharing receipts and related attachments with their company. I'll treat the related in this issue similarly.

**Comment by Beamanator** [comment]
A few updates now that I'm back from OOO :D

Updates:
- iOS / Android:
    - Upstream PR: https://github.com/DylanVann/react-native-fast-image/pull/953
        - No movement for a few weeks
        - I reached out to Dylan via email on his GH profile (following copy from [this SO](https://stackoverflowteams.com/c/expensify/questions/12761))
    - Draft App PR: https://github.com/Expensify/App/pull/13304
- Web / Desktop:
    - Upstream PR: https://github.com/necolas/react-native-web/pull/2442
        - Still no movement in upstream PR
    - Expensify Fork PR: https://github.com/Expensify/react-native-web/pull/8
        - I did a review, but i'm asking Marc for a review as well & asking if he thinks we want another eng or two to check if out before we proceed, since it's kinda unfamiliar territory
    - App PR (just a draft / in test state): https://github.com/Expensify/App/pull/13036/files
- Multiple avatar image types
    - https://github.com/Expensify/App/issues/13038 is now on hold for WAQ - though, since WAQ is winding down, I might still try to make progress this week
    - Bondy gave a nice recap of where we left off in our last slack convo here: https://github.com/Expensify/App/issues/13038#issuecomment-1365292436
- Default avatars:
    - No updates, Georgia has been OOO

New issues I think we want to start tracking here:
- https://github.com/Expensify/App/issues/12893
    - As mentioned [here](https://github.com/Expensify/App/issues/12893#issuecomment-1363228637), we are putting this on hold for now since we don't have clear next steps, and there's more high priority issues (above)
- https://github.com/Expensify/App/issues/13585
- https://github.com/Expensify/App/issues/13910

**Comment by trjExpensify** [comment]
On these two new ones: 

>mWeb - Chat - The avatar in the group's welcome message is not displayed correctly #13585

It's specifically in the case of a transparent background PNG, but I don't quite understand how that one is a bug. I've commented in the issue. 

>Profile images appear in square when gif is uploaded as profile pic #13910
This one won't be an issue after #13038, right?

**Comment by Beamanator** [comment]
Arg will send update tomorrow 🙃

**Comment by Beamanator** [comment]
> > mWeb - Chat - The avatar in the group's welcome message is not displayed correctly #13585
> 
> It's specifically in the case of a transparent background PNG, but I don't quite understand how that one is a bug. I've commented in the issue.

Agreed, looks like we have a resolution here in the issue

> > Profile images appear in square when gif is uploaded as profile pic #13910
> 
> This one won't be an issue after #13038, right?

It shouldn't be, correct 👍 

And here's anotha few image-related issues I added above! Neither of these are on hold since they don't seem to be related to caching:
- https://github.com/Expensify/App/issues/13962
- https://github.com/Expensify/App/issues/14027

**Comment by Beamanator** [comment]
Updates:

- iOS / Android:
    - Upstream PR: https://github.com/DylanVann/react-native-fast-image/pull/953
        - Still no movement, I reached out last week and didn't hear back
    - Patch PR: https://github.com/Expensify/App/pull/13304
        - Has undergone a lot of reviews already, seems we're still trying to nail down the last few little flicker issues then we can merge!
- Web / Desktop
    - Upstream PR: https://github.com/necolas/react-native-web/pull/2442
        - Still no movement since Dec 2, 2022
    - Fork PR: https://github.com/Expensify/react-native-web/pull/8
        - Has had quite a few review comments already, basically trying to limit the amount of changes since this library's codebase is pretty unfamiliar to the reviewers so far.
        - Kidroca has a second draft PR we will review as it has far fewer code changes: https://github.com/Expensify/react-native-web/pull/11/files
    - Draft App PR: https://github.com/Expensify/App/pull/13036

**Comment by trjExpensify** [comment]
>Neither of these are on hold since they don't seem to be related to caching:

Seems like https://github.com/Expensify/App/issues/14027 is on hold for the SVG default avatars PR. Should it not be?

**Comment by Beamanator** [comment]
@trjExpensify I think it make sense to be on hold for the SVG default avatars - as mentioned [here](https://github.com/Expensify/App/issues/14027#issuecomment-1374864993):

> this issue only happened with the users that didn't have avatar. Since I changed avatar of the user inside group that flickers, this issue didn't happen anymore.

**Comment by trjExpensify** [comment]
Okay cool, so I got confused with the "neither of these are on hold" part of the comment.

**Comment by Beamanator** [comment]
Oh i see the confusion, that's my bad 😅

**Comment by trjExpensify** [comment]
All good! Can we get another run down of the latest here for this week? Native caching PR is sooo close, really looking forward to getting that across the line.

**Comment by Beamanator** [comment]
Native caching PR has been merged! Default avatars is also veryyyyy close, will update this tomorrow :D

**Comment by Beamanator** [comment]
Updates:
- iOS / Android
    - Commented in [Upstream PR](https://github.com/DylanVann/react-native-fast-image/pull/953), tagging maintainer to try to get more attention
    - [Caching PR](https://github.com/Expensify/App/pull/13304) workaround merged! 
        - _Don't get excited, we had regressions last time so lets wait and see how this goes 🙏 _
    - Issue created for me to keep track of upstream PR: https://github.com/Expensify/App/issues/14382
- Web / Desktop
    - 
- Multiple avatar image types:
    - 
- Default avatars
    -

**Comment by JmillsExpensify** [comment]
Dang, making some really great progress here! Let's close the loop on default avatars. That's an exciting polish item we want to get in the hands of members.

**Comment by trjExpensify** [comment]
Soooo close on the caching PRs. Very excited!

**Comment by Beamanator** [comment]
Updates:
- iOS / Android
    - No reaction to my comment in [Upstream PR](https://github.com/DylanVann/react-native-fast-image/pull/953).
        - Will leave further updates in this issue: https://github.com/Expensify/App/issues/14382
    - [Caching PR](https://github.com/Expensify/App/pull/13304) still looking good!  Has been in production for 4 days now 👍 
- Web / Desktop
    - My comment in [Upstream PR](https://github.com/necolas/react-native-web/pull/2442) from last week got the maintainer's attention, which is great 👍 
    - [Fork PR](https://github.com/Expensify/react-native-web/pull/11) is looking great! I gave my 👍 of approval, waiting for maybe 1 more reviewer before merging tomorrow (if there's no additional concerns)
- Multiple avatar image types:
    - Seems like we might go with being "ok" that a few image types don't work on some platforms, and we'll create an issue aimed at the long-term fix in the meantime
- Default avatars
    - [PR](https://github.com/Expensify/App/pull/13216) - looks like Georgia has been ooo a bit, there's a decent amount of conflicts in her PR since there's so many files touched 😬

**Comment by Beamanator** [comment]
Quick little update: the React native web fix has been merged! I'm trying to publish our update (following [this SO](https://stackoverflowteams.com/c/expensify/questions/13855))

**Comment by trjExpensify** [comment]
Sahweet! So we now need to the App PR to get through review, before we can retest the held issues. Is that correct?

**Comment by Beamanator** [comment]
Updates:
- iOS / Android
    - Considering this ✅ 
- Web / Desktop
    - [Upstream PR](https://github.com/necolas/react-native-web/pull/2442): Kidroca is working on updating w/ what we merged & some tests, requested by the maintainer
    - [Fork PR](https://github.com/Expensify/react-native-web/pull/11) is merged!
    - [App PR](https://github.com/Expensify/App/pull/13036) just needs testing & approval 🙏  - there's some weird CORS issues also being handled at the same time, hopefully that won't hold us up long
- Multiple avatar image types:
    - Made a new issue for gif & bmp fixes: https://github.com/Expensify/App/issues/14751
    - Planning to repurpose https://github.com/Expensify/App/issues/13038 to fix SVG avatars
- Default avatars
    - [PR](https://github.com/Expensify/App/pull/13216) was merged! 💪

**Comment by MelvinBot** [comment]
@Beamanator, @trjExpensify Eep! 4 days overdue now. Issues have feelings too...

**Comment by Beamanator** [comment]
@trjExpensify I'm trying to figure out if the CORS issue is being handled / discussed elsewhere - I remember a slack convo somewhere but we need to bring this convo to Slack in my opinion, to get more advice from more knowledgable people (at least related to CORS & the issue we're hitting). So I bumped kidroca in the PR 👍

**Comment by Beamanator** [comment]
Also I think it makes sense to change this priority to weekly since we don't need daily updates at the minute, right?

**Comment by trjExpensify** [comment]
Okay, I'm fine with that, as long as the individual `bug` issues are on `daily` still as per the process.

**Comment by Beamanator** [comment]
Updates:
- Web / Desktop
    - [Upstream PR](https://github.com/necolas/react-native-web/pull/2442): was updated last week, waiting on maintainer reviews
    - [App PR](https://github.com/Expensify/App/pull/13036): looks like we need to resolve some CORS issues (being discussed [here](https://github.com/Expensify/App/issues/14227)) in order to move forward with this
- Multiple avatar image types:
    - I split the gif & bmp fix issue into a new one specifically for `bmp` fix since the fixes are not related (I should have done that earlier).
    - So the 3 issues for this are:
        - https://github.com/Expensify/App/issues/14751
        - https://github.com/Expensify/App/issues/15035
        - https://github.com/Expensify/App/issues/13038

**Comment by trjExpensify** [comment]
How we looking here, @Beamanator? CORS issue linked is closed, so we're unblocked again on that App PR?

**Comment by Beamanator** [comment]
Trying to find someone to take over https://github.com/Expensify/App/issues/14991 - [posted in Slack](https://expensify.slack.com/archives/C01GTK53T8Q/p1677252475615489)

**Comment by trjExpensify** [comment]
@Beamanator what's the latest? 

>Looks like the CORS issue is now being discussed here: https://github.com/Expensify/App/issues/15087 so App PR is not unblocked

The [linked issue](https://github.com/Expensify/App/issues/15087#issuecomment-1457677107) seems to have a PR moving on it again, so is web image caching going to be unblocked by that or still conditional on #14991?

**Comment by trjExpensify** [comment]
So I'm a bit confused by what each issue is doing then if that's the case. 😕

**Comment by trjExpensify** [comment]
Bump, @Beamanator? 

or perhaps @mountiny can help enlighten me here as the PR linked to #15087 is done. Does that unblock https://github.com/Expensify/App/pull/13036 for web/desktop image caching?

**Comment by Beamanator** [comment]
Ya sryyyyy

So I really don't believe THAT change is what we're holding on, it's actually this one: https://github.com/Expensify/App/issues/14991

Which was moving, but as of [this](https://expensify.slack.com/archives/CC7NECV4L/p1679342455911239?thread_ts=1678994690.390109&cid=CC7NECV4L) whatsnext post, we may have a big upcoming change that could help us a lot

**Comment by mountiny** [comment]
What Alex said, we got a bit blocked with the headers changes but moving to our own servers should help a lot

**Comment by trjExpensify** [comment]
Alright, so can we update this PR and associated issue to note what it's actually held on, because #14991 has merged and that's not it? https://github.com/Expensify/App/pull/13036

**Comment by trjExpensify** [comment]
^^ @Beamanator @mountiny bump on the above.

**Comment by Beamanator** [comment]
@trjExpensify sorry for the delay!

Sorry but what's "this PR" that you're referring to [here](https://github.com/Expensify/App/issues/10894#issuecomment-1488547681)?

My understanding is that we're still waiting on [moving NewDot web hosting to our servers](https://expensify.slack.com/archives/CC7NECV4L/p1679342455911239?thread_ts=1678994690.390109&cid=CC7NECV4L) (mentioned [here](https://github.com/Expensify/App/issues/14991#issuecomment-1480272712))

**Comment by Beamanator** [comment]
Aahhh ok ok ok cool yeah sorry for not being clear about https://github.com/Expensify/App/issues/14991 being on hold, i justtttt put a hold in the title

**Comment by neil-marcellini** [comment]
So what issue is currently being worked on for this? It's all very confusing.

**Comment by JmillsExpensify** [comment]
Is the server migration actively being worked on? Where can I follow that issue/project?

**Comment by Beamanator** [comment]
Good question! I just ping'd Amy & Marco so I'll report back when I hear :D

**Comment by JmillsExpensify** [comment]
Yeah that's kind of my concern. The WN thread also looks very inconclusive as to whether we agreed it should be worked on.

**Comment by Beamanator** [comment]
Looks like Justin just created a tracking issue for the project: https://github.com/Expensify/Expensify/issues/277714

**Comment by Beamanator** [comment]
Still pretty on hold based on tracking issue

**Comment by JmillsExpensify** [comment]
Should we put this issue on hold for the E/E issue?

**Comment by Beamanator** [comment]
@JmillsExpensify done 👍

**Comment by trjExpensify** [comment]
Works for me!

**Comment by trjExpensify** [comment]
Justin has the HL doc out for review, but we're still a little while off unholding this.

**Comment by Beamanator** [comment]
Still on hold - HL of doc is still in review

**Comment by trjExpensify** [comment]
Migrating servers to S3 is progressing, but this is still on hold!

**Comment by trjExpensify** [comment]
No change. Migrating servers to S3 is progressing, but this is still on hold!

**Comment by trjExpensify** [comment]
Still in the works!

**Comment by roryabraham** [comment]
Just noticed that https://github.com/ascoders/react-native-image-zoom, the library we use for native `ImageView` has been abandoned.

**Comment by trjExpensify** [comment]
Server migration still in the works. 

As for Rory's [comment](https://github.com/Expensify/App/issues/10894#issuecomment-1603070604), @Beamanator what are your thoughts on that? The remaining items we have are held on image caching, but replacing the use of an abandoned library seems unrelated. Is that something we need to explore now?

**Comment by trjExpensify** [comment]
@Beamanator bump on this!

>As for Rory's https://github.com/Expensify/App/issues/10894#issuecomment-1603070604, @Beamanator what are your thoughts on that? The remaining items we have are held on image caching, but replacing the use of an abandoned library seems unrelated. Is that something we need to explore now?

**Comment by roryabraham** [comment]
Asked in slack about this: https://expensify.slack.com/archives/C01GTK53T8Q/p1689185793050179

**Comment by trjExpensify** [comment]
Server migration still in the works. 

As for the RN image zoom library, seems like something Margelo is working on could allow us to switch it out, but it isn't causing a problem as is.

**Comment by trjExpensify** [comment]
No change.

**Comment by Beamanator** [comment]
no change

**Comment by trjExpensify** [comment]
Still blocked on the server migration project dependency.

**Comment by trjExpensify** [comment]
Still blocked on the server migration project dependency.

**Comment by trjExpensify** [comment]
Still blocked on the server migration project dependency.

**Comment by trjExpensify** [comment]
Still blocked on the server migration project dependency.

**Comment by Beamanator** [comment]
still blocked

**Comment by trjExpensify** [comment]
[Justin is asking for a summary](https://github.com/Expensify/App/issues/14991#issuecomment-1743750903) on the blocking issue to validate if there's another path without a server migration.

**Comment by trjExpensify** [comment]
Checked in on that.

**Comment by trjExpensify** [comment]
The first step in the dependency for the [CORS issue](https://github.com/Expensify/App/issues/14991#issuecomment-1781234670) is pretty much over the line (awaiting a confirmation), so we should be able to come back to the [image caching issue now](https://github.com/Expensify/App/issues/12603) to unblock the rest.

**Comment by trjExpensify** [comment]
Image caching PR is in review. Getting there!

**Comment by trjExpensify** [comment]
Image caching PR is progressing still!

**Comment by trjExpensify** [comment]
Super close. PR merged, so after the deploy cycle we should be able to come back here!

**Comment by trjExpensify** [comment]
Image caching PR was deployed and reverted. Awaiting this now before re-implementing: https://github.com/Expensify/App/issues/32703

**Comment by Beamanator** [comment]
same ^

**Comment by trjExpensify** [comment]
Same, awaiting this now before re-implementing: https://github.com/Expensify/App/issues/32703

**Comment by blimpich** [comment]
https://github.com/Expensify/App/issues/32703 is fixed, should be good to move forward now. Thank you for your patience.

**Comment by trjExpensify** [comment]
In light of #32703 being fixed, the [new image caching PR](https://github.com/Expensify/App/pull/34505) is in review.

**Comment by trjExpensify** [comment]
https://github.com/Expensify/App/pull/34505 is in review still!

**Comment by trjExpensify** [comment]
The image caching PR was [reverted again](https://github.com/Expensify/App/pull/35636), so we're back to figuring that out in #12603.

**Comment by trjExpensify** [comment]
[New PR](https://github.com/Expensify/App/pull/36560) for #12603 is up.

**Comment by trjExpensify** [comment]
Plan being worked on!

**Comment by trjExpensify** [comment]
https://github.com/Expensify/App/pull/36560 in the works!

**Comment by trjExpensify** [comment]
We merged it very recently. Looking good so far. So hopefully we can turn to https://github.com/Expensify/App/issues/9402 soon.

**Comment by mallenexpensify** [comment]
Updated title to remove the hold cuz the issue it was held on hit production
https://github.com/Expensify/App/issues/12603
(I have another issue that's held on this one)

**Comment by trjExpensify** [comment]
What issue is it, @mallenexpensify?

**Comment by trjExpensify** [comment]
Interesting! I don't think that needs to hold on this issue, it looks like it's being addressed here: https://github.com/Expensify/App/issues/35502

I'm going to go ahead and close this out now. We aren't adding anything new to it or require it for CAP or anything. The two remaining items we have are [here](https://github.com/Expensify/App/issues/9402) and [here](https://github.com/Expensify/App/issues/4629), so follow along there if anyone is still interested!
